python/dllib/src/bigdl/dllib/feature/dataset/news20.py

```diff
@@ -28,26 +28,24 @@
 def download_news20(dest_dir):
     file_name = "20news-18828.tar.gz"
     file_abs_path = base.maybe_download(file_name, dest_dir, NEWS20_URL)
-    tar = tarfile.open(file_abs_path, "r:gz")
-    extracted_to = os.path.join(dest_dir, "20news-18828")
-    if not os.path.exists(extracted_to):
-        print("Extracting %s to %s" % (file_abs_path, extracted_to))
-        tar.extractall(dest_dir)
-        tar.close()
-    return extracted_to
+    with tarfile.open(file_abs_path, "r:gz") as tar:
+        extracted_to = os.path.join(dest_dir, "20news-18828")
+        if not os.path.exists(extracted_to):
+            print("Extracting %s to %s" % (file_abs_path, extracted_to))
+            tar.extractall(dest_dir)
+        return extracted_to
```

Comment:
return needs to be put outside with?
Otherwise after the return, the close function has no chance to be invoked?

Reply:
Based on my before simple test, the close function will be invoked.
But for safety, have moved the return outside of 'with'.